src/app/api/quality_matrix/collections.py

```diff
@@ -66,8 +66,29 @@ async def id_to_title_mapping(node_id: uuid.UUID) -> dict[str, str]:
 
 async def collection_quality(
     node_id: uuid.UUID, match_keyword: str = "path"
-) -> QUALITY_MATRIX_RETURN_TYPE:
+) -> tuple[list[QualityOutput], {dict[str, int]}]:
     mapping = await id_to_title_mapping(node_id)
     columns = queried_collections(node_id)
     properties = get_properties()
-    return await _quality_matrix(columns, mapping, match_keyword, node_id, properties)
+    _quality = await _quality_matrix(
+        columns, mapping, match_keyword, node_id, properties
+    )
+    _quality = transpose(_quality, [name for name in mapping.values()])
+    return _quality, {prop: 0 for prop in properties}
+
+
+def transpose(entries: list[QualityOutput], columns: list[str]) -> list[QualityOutput]:
```

Comment:
AFAIK, `QualityOutput` should not be used for both the input and return type of that function? Reasoning: `QualityOutput` uses `metadatum` as member variable, but the transposed return would have e.g. collections assigned to the metadatum?

Reply:
Refactored according to [the following comment](https://github.com/openeduhub/metaqs-main/pull/70#discussion_r930993225)